BASE/src/Microsoft.ApplicationInsights/Extensibility/TelemetryConfiguration.cs

```diff
@@ -334,6 +335,11 @@ public string ConnectionString
         /// </summary>
         public TelemetrySink DefaultTelemetrySink => this.telemetrySinks.DefaultSink;
 
+        /// <summary>
+        /// Gets an envelope for Azure.Core.TokenCredential which provides an AAD Authenticated token.
+        /// </summary>
+        public CredentialEnvelope CredentialEnvelope { get; private set; }
```

Comment:
Consider making this internal, if end users are not expected to read this back.

Reply:
@pharring, We're planning on making the CredentialEnvelope `internal`.
We want to get your library added to our InternalsVisibleTo [here](https://github.com/microsoft/ApplicationInsights-dotnet/blob/ddd41c578c6cfad0788bf1defa11e68792cc35c5/BASE/src/Microsoft.ApplicationInsights/Properties/AssemblyInfo.cs#L7-L11).

Reply:
@xiaomi7732 for https://github.com/microsoft/ApplicationInsights-Profiler-AspNetCore

Snapshot Collector is Microsoft.ApplicationInsights.SnapshotCollector, but note that it has a different public key.

Reply:
Thanks for tagging me on this. AI.SDK and Profiler have different build cycles - and we have mixed references. **How are we going to make that work?** For example, our desktop build (no StrongName) assemblies might reference public/stable version of app insights SDK, in which case, the InternvalVisibileTo (to a strongNamed assembly) won't work. What is that I missed?

Reply:
I don't think you need to worry about version numbers. InternalsVisibleTo uses just the assembly's name and public key. The question really is whether you expect to take a dependency on a version of the core SDK that has CredentialEnvelope, or would you stick with an older version and use reflection to get CredentialEnvelope.

I still haven't decided for Snapshot Collector. There's going to be some reflection anyway in order to use the CredentialEnvelope.

Reply:
On this topic, I think I am asked to provide assembly names and public keys so that in Core SDK, it knows to expose the internal classes.

The deal is, during the development, our assemblies aren't going to be strongnamed - that means we won't be able to see the internal classes in the Core SDK anyway.

On the other hand, reflection will work, with or without `InternalVisibleTo`.

So, what is the point to add the `InternalVisibleTo` for our assemblies? And that is what I am asking, in case I missed something.

Reply:
We're changing this property to `internal`. 
Optionally, if you want to utilize this without reflection you can submit a PR to add your SDK to our `InternalsVisibleTo`.